libdnf/transaction/Swdb.cpp

```diff
@@ -77,9 +78,14 @@ Swdb::Swdb(const std::string &path)
 void
 Swdb::resetDatabase()
 {
+    auto dbpath=getPath().c_str();
     conn->close();
-    if (pathExists(getPath().c_str())) {
-        remove(getPath().c_str());
+    if (pathExists(dbpath)) {
+        if (remove(dbpath)) {
+            throw std::runtime_error(
+                tfm::format(_("Failed to remove history database file '%s': '%s'"),
+                            dbpath, strerror(errno)));
```

Comment:
We need to move away from `std::runtime_error`. See my PR: https://github.com/rpm-software-management/libdnf/pull/866, in particular The `Error` class in the first commit: https://github.com/rpm-software-management/libdnf/pull/866/commits/baf6a9d8a8f3a64b2e242146340a4024f2f0cd70

Any comments on my PR are welcome :slightly_smiling_face:

Until my PR gets merged treat this as a note only, when it gets merged you can change this to the `Error` class. That will make DNF not print a traceback but only the actual error message, which is correct in this case, since the file not being removable is out of our control and the message itself is sufficient, a traceback should not be needed.